src/main/java/org/jitsi/srtp/crypto/OpenSslHmacSpi.java

```diff
@@ -25,15 +27,35 @@
  * @author Lyubomir Marinov
  */
 public class OpenSslHmacSpi
-    extends MacSpi
+    extends MacSpi implements AutoCloseable
 {
+    private static class OpenSslHmacSpiCleanable implements Runnable
+    {
+        /**
+         * The context of the OpenSSL (Crypto) library through which the actual
+         * algorithm implementation is invoked by this instance.
+         */
+        long ptr;
+
+        @Override
+        public void run()
+        {
+            long ptrCopy = ptr;
+            if (ptrCopy != 0)
+            {
+                ptr = 0;
+                OpenSslHmacSpi.HMAC_CTX_destroy(ptrCopy);
```

Comment:
True, I added `synchronized` to properly guard this. I'm not experienced enough with the Cleaner if that's an acceptable solution...

Reply:
I just saw that there's a mistake anyway: always creating a new cleaner (which AFAIU creates a new thread every time). I'll try to tidy that up.

I'm not really sure about the double-call guard yet. Is that guaranteed somewhere?

Reply:
The doc for `PhantomCleanable::clean()` says 
```
    /**
     * Unregister this PhantomCleanable and invoke {@link #performCleanup()},
     * ensuring at-most-once semantics.
     */
```
which I think means it does the right thing.

Reply:
Out of curiosity, is there a reason you did the `Cleanable` change for `OpenSslHmacSpi` but not `OpenSslAesCipherSpi` or `OpenSslAesGcmAuthOnlyCipherSpi`?   Those also have `finalize` methods.

Reply:
> Out of curiosity, is there a reason you did the `Cleanable` change for `OpenSslHmacSpi` but not `OpenSslAesCipherSpi` or `OpenSslAesGcmAuthOnlyCipherSpi`? Those also have `finalize` methods.

I did the finalizer change because I was annoyed by the deprecation warning during compilation and missed the other two. Fixed now.